### PR TITLE
Added dominos coupons command

### PR DESCRIPTION
**Pull request by 49Indium**
Added the previously unimplemented dominoscoupons command. Returns a list of dominos coupons from [couponese](https://www.couponese.com/store/dominos.com.au/). Heavily *inspired* by @bradleysigma's previous implementation for the slack bot.

Also completes part of #1.

### CODE REVIEW

File: uqcsbot/dominos_coupons.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            logging.warning(
+                f"Could not connect to dominos coupon site ({COUPONESE_DOMINOS_URL}): {error.response.content}"
+            )
+            await interaction.edit_original_response(
+                content=f"Sadly could not reach the coupon website (<{COUPONESE_DOMINOS_URL}>)..."
+            )
+            return
+        except HTTPResponseException as error:
+            logging.warning(
+                f"Received a HTTP response code that was not OK (200), namely ({error.http_code}). Error information: {error.message}"
```
Comment (andrewj-brown) [inline uqcsbot/dominos_coupons.py:58]: As the log format is somewhat horrific, it'd be kinda nice to keep logs as terse as possible - saying "response was not OK" is implied by giving a non-200 error code later in the sentence.

File: uqcsbot/dominos_coupons.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                ]
+            )
+        except ValueError:
+            return True
+
+    def keyword_matches(self, keyword: str) -> bool:
+        return keyword.lower() in self.description.lower()
+
+
+class HTTPResponseException(Exception):
```
Comment (andrewj-brown) [inline uqcsbot/dominos_coupons.py:101]: Personal preference: putting exception types at the top of the file keeps them all in one spot and highlights them for anyone who's quickly skimming.

File: uqcsbot/dominos_coupons.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            return
+
+        message = "Domino's Coupons:\n"
+        for coupon in coupons:
+            message += f"`{coupon.code}` - {coupon.description} *[Expires: {coupon.expiry_date}]*\n"
+        await interaction.edit_original_response(content=message)
+
+
+class Coupon:
+    def __init__(self, code: str, expiry_date: str, description: str) -> None:
```
Comment (andrewj-brown) [inline uqcsbot/dominos_coupons.py:78]: I question the utility of adding a coupon class when you're just manually accessing the fields anyway, but I guess there's not really a better way without getting into horrific kwarg-generator bullshit for `embed.add_field`.

I guess you could have a method that chains embeds (i.e. something like `embed = Coupon.add_self_to_embed(embed)`) but that's up to you.

**Review by andrewj-brown — COMMENTED** [review]
Couple minor points - nothing blocking merge. Git seems to have done a funny and not included your 3 latest commits into this PR; I was unable to rectify that by opening a replacement (because this one is still linked).

So I guess whoever merges this, beware and make sure all `dominos` commits are included, which may require closing this PR and opening a new one to merge.

File: uqcsbot/dominos_coupons.py (hunk 3 above)
Comment (49Indium) [reply uqcsbot/dominos_coupons.py:97]: I'm going to skip on it. The class is definitely overkill, but it seems better than passing a dictionary around. Also, it is only put into an embed once, so I'll leave creating that method to someone else.

**Review by JamesDearlove — APPROVED** [review]
Seems fine to me, lgtm

**Pull request merged**